### DISCUSSION TITLE
PRQL Editor and SQL Notebook

### DISCUSSION THREAD

**Discussion by ryanhamilton** [PRQL/prql#5046] [Show and tell]
Last year we added PRQL support to QStudio: 
https://www.timestored.com/qstudio/prql-ide
You can use PRQL to query 30+ databases and export or chart the results easily.

A member of your community kindly wrote a quick start guide:
https://github.com/richb-hanover/qStudio-PRQL_Quick_Start

I just wanted to update you that QStudio now supports creating interactive notebooks:
https://www.timestored.com/sqlnotebook/

It would be great to have more PRQL users, Rich has provided a lot of ideas and great input
but really we need more users to justify investing more time to improve the integration. 

![prql-sql-ide](https://github.com/user-attachments/assets/d226d43f-f7d3-40d0-aab8-91a0ec1913c9)

I hope you find it useful. Thanks,
Ryan

**Comment by richb-hanover** [comment]
Thanks, Ryan. I have found the combination of QStudio and PRQL to be enormously productive.

As a die hard Mac user, I also created a macOS application that combines QStudio and the `prqlc` binary into a double-clickable program. You can get it from: https://randomneuronsfiring.com/wp-content/uploads/QStudio.zip

**Comment by ryanhamilton** [comment]
Thanks to Rich work I created an official mac download:
https://www.timestored.com/qstudio/download
See the qstudio-mac.zip which should have an App inside.

**Reply by richb-hanover** [reply]
Yes - _qstudio-mac.zip_ does contain a macOS app. 

But it doesn't seem to have the latest versions of either QStudior or prql... (It has 4.06 & 0.13.2. I believe the latest is 4.12 for QStudio and 0.13.7 for prqlc) The [create-qstudio-macos-app](https://github.com/richb-hanover/create-qstudio-macos-app/blob/main/create-qstudio-macos-app.sh) script has code to download the latest binaries (the version numbers are constants at the top of the script).

Do you have access to a Linux (or macOS) box to run the script? I've tested it on both. Thanks again
